fmriprep/workflows/bold/base.py

```diff
@@ -279,11 +278,23 @@ def init_func_preproc_wf(bold_file, ignore, freesurfer,
                 't1_2_fsnative_forward_transform', 't1_2_fsnative_reverse_transform']),
         name='inputnode')
 
-    if isinstance(bold_file, list):
-        inputnode.inputs.iterables = ("bold_file", bold_file)
+    if multiecho:  # if multi-echo data
+        # iterate most preprocessing steps over each echo
+        inputnode.iterables = ('bold_file', bold_file)
```

Comment:
Could you check if we can completely remove this if .. else?:

1. Check that one can set `inputnode.inputs.bold_file = bold_file` for all workflows, and then
2. Check that setting `inputnode.iterables = ('bold_file', bold_file)` shadows the setting in 1).

I don't think we have used that approach ever, so I don't know how nipype will understand that. But, if it works, then the iterables could be set within the multiecho block in line 460.

Reply:
I would never have thought of that, but it seems to have worked ! 🎉